src/DataModel/IResource.ts

```diff
@@ -0,0 +1,5 @@
+export interface IResource
+{
+    "@id": string;
+    "@type"?: Array<string>
```

Comment:
Why do we actually use properties here and not methods? The programming interface doesn't need to match what's send on the wire